File: 002_pilas.js
//Implementar la estructura de datos: Pila.

class Pilas{ //Creamos un objeto llamado pila.
	constructor(){ //El método constructor no recibe parametros.
		this.pila = []
  }

	cantidad(){ //Tiempo de ejecución O(1).
		return len(this.pila) //Método que nos indica la cantidad de elementos dentro de la pila.
  }

	estaVacio(){ //Tiempo de ejecución O(1).
		return len(this.pila) == 0 //Devuelve verdadero si el tamaño es igual a cero.
  }

	apilar(elemento){ //Tiempo de ejecución O(1).
		this.pila.push(elemento) //El nuevo elemtno es almacenado al fina de la pila.
  }

	extraer(){ //Tiempo de ejecución O(1).
		if (this.estaVacio()){ 
			return  //Si la lista esta vacia nos saca de la ejecución del programna.
    }
		return this.pila[-1] //En caso contrario nos devuelve el ultimo elemento en la pila.
    
  }
	
	remover(){ //Tiempo de ejecución O(1).
		if (this.estaVacio()){ 
			return  //Si la lista esta vacia nos saca de la ejecución del programna.
    }
		return this.pila.pop() //Devuelve el ultimo elemento en la pila y lo elimina de la misma.
  }

  mostrarLista(){ //Tiempo de ejecución O(n)
		if (this.pila == null){ //Si la lista esta vacia se termina la ejecución del programa.
			return }
		else{ //De otra manera se recorre la lista y se imprimen cada uno de sus elementos.
			for (var i=0; i < this.pila.length; i++ ){        
				textAlign(CENTER)        
        text(this.pila[i], 100, i*30)
        ellipse(100, i*30, 30,30)  
      }
			
      }     
    } 
      


} // cierra la clase pilas

// Visualizar resultados en el navegador. 
var p 
function setup() {
  createCanvas(800, 400) // Definir las dimensiones de la pantalla.
  p = new Pilas()
  
  for (var i = 0; i < 10; i++) {
    p.apilar(floor(random(0, 100)))
  }
  console.log( p.pila ) 
}


function draw() {
  background(0) // El color de fondo es negro.
  stroke(255) // El grosor del trazo.
  noFill() // La figura no tiene relleno.
   p.mostrarLista()
}
	

File: 002_pilas.py
#Implementar la estructura de datos: Pila.
import numpy as np

class Pilas: #Creamos un objeto llamado pila.
	def __init__(self): #El método constructor no recibe parametros.
		self.pila = []
		

	def cantidad(self): #Tiempo de ejecución O(1).
		return len(self.pila) #Método que nos indica la cantidad de elementos dentro de la pila.

	def estaVacio(self): #Tiempo de ejecución O(1).
		return len(self.pila) == 0 #Devuelve verdadero si el tamaño es igual a cero.

	def apilar(self, elemento): #Tiempo de ejecución O(1).
		self.pila.append(elemento) #El nuevo elemtno es almacenado al fina de la pila.

	def extraer(self): #Tiempo de ejecución O(1).
		if self.estaVacio(): 
			return  #Si la lista esta vacia nos saca de la ejecución del programna.
		return self.pila[-1] #En caso contrario nos devuelve el ultimo elemento en la pila.
	
	def remover(self): #Tiempo de ejecución O(1).
		if self.estaVacio(): 
			return  #Si la lista esta vacia nos saca de la ejecución del programna.
		return self.pila.pop() #Devuelve el ultimo elemento en la pila y lo elimina de la misma.


if __name__ == "__main__":
	vector = np.random.randint(100,size=6)
	p = Pilas()
	print("Cantidad de elementos al inicio: ", p.cantidad())
	print("Esta la pila vacia: ", p.estaVacio())
	
	#Insertar valores en la pila
	for i in vector:
		p.apilar(i)
		
	print("Elementos dentro de la pila\n", p.pila)
	print("Extraer el ultimo elemento ingresado\n", p.extraer())
	print("La pila no cambia solo muestra el elemento\n", p.pila)
	print("Remover el ultimo elemento ingresado\n", p.remover())
	print("La pila si cambia al remover el elemento\n",p.pila)